modules/compatibility-tests/src/jobs/java/org/apache/ignite/internal/compute/SendAllMetastorageCommandTypesJob.java

```diff
@@ -67,7 +76,13 @@ private static CompletableFuture<Void> sendCompactionCommand(MetaStorageManagerI
             Method sendCompactionCommand = metastorage.getClass().getDeclaredMethod("sendCompactionCommand", long.class);
             sendCompactionCommand.setAccessible(true);
 
-            return (CompletableFuture<Void>) sendCompactionCommand.invoke(metastorage, metastorage.appliedRevision());
+            // This is effectively a no-op (applied revision is not 0 at this point), we are waiting just in case to increase test
+            // robustness.
+            while (metastorage.appliedRevision() == 0) {
```

Comment:
I don't insist, but maybe it would be better to do it asynchronously via `org.apache.ignite.internal.metastorage.impl.MetaStorageManagerImpl#registerRevisionUpdateListener`. It's up to you.

Reply:
Changed the current approach to be less hacky